File: 2256-minimum-average-difference/2256-minimum-average-difference.py
class Solution:
    def minimumAverageDifference(self, nums: List[int]) -> int:
        totalsum = 0
        currentsum = 0
        n = len(nums)
        minsum = math.inf
        for i in range(n):
            totalsum += nums[i]
        for i in range(n):
            currentsum += nums[i]
            
            fpavg = currentsum
            fpavg//=(i+1)
            spavg = totalsum - currentsum
            if(i == n-1):
                if(fpavg<minsum):
                    return n-1
                else:
                    break
            spavg //= (n-i-1)
            diff = abs(fpavg -spavg)
            if(diff<minsum):
                minsum = diff
                answerIndex = i
        return answerIndex

File: 2256-minimum-average-difference/2256-minimum-average-difference.cpp
class Solution {
public:
    int minimumAverageDifference(vector<int>& nums) {
        long totalsum = 0, currentsum = 0;
        int n = nums.size();
        for(int i = 0; i<n;i++){
            totalsum += nums[i];
        }
        int minsum = INT_MAX;
        int answerIndex = 0;
        for(int i = 0; i< n; i++){
            currentsum += nums[i];
            long fpavg = currentsum/(i+1);
            
            
            if(i == n-1){
                if(fpavg<minsum)
                    return n-1;
                else
                    break;
            }
            long spavg = (totalsum - currentsum)/(n-i-1);
            long diff = abs(fpavg - spavg);
            if(diff< minsum){
                minsum = diff;
                answerIndex = i;
            }
        }
        return answerIndex;
    }
};